### PR TITLE
Improve diff in bank.tabbankaccount

### LINKED ISSUE #310: Fix the diff on bank.tabbankaccount
Match on Bilagsnr, Fakturanr and KID.

### PR DESCRIPTION
**Pull request by agvald**
Closes #310

### CODE REVIEW

File: modules/bank/model/bank.class.php [hunk 1]
```diff
@@ -194,7 +194,10 @@ private function _get_bank_voting_tilbake($Period) {
             if($this->type == 'voucher' && round($row_voucher->{$this->side},2) == round($this->searchstring,2) && round($this->searchstring, 2) > 0)
                 $bankvoting_tilbake_hash[$id]->{'class' . $this->side} = 'number red';
 
-            $this->set_closeable_vouchertilbake(0, $row_voucher->KID, $row_voucher->InvoiceNumber, - $row_voucher->AmountIn + $row_voucher->AmountOut, 'bank voting tilbake');
+            $row_voucher->KID           = trim($row_voucher->KID);
+            $row_voucher->InvoiceNumber = trim($row_voucher->InvoiceNumber);
+            $row_voucher->JournalID     = trim($row_voucher->JournalID);
+            $this->set_closeable_vouchertilbake(0, $row_voucher->KID, $row_voucher->InvoiceNumber, (-$row_voucher->AmountIn + $row_voucher->AmountOut), 'bank voting tilbake', $row_voucher532>JournalID);
```
Comment (mladjo2505) [inline modules/bank/model/bank.class.php]: `$row_voucher532->` change to `$row_voucher->`

**Comment by mladjo2505** [comment]
On all view pages for bank add a check for read only user and check for fakturabank import access.

File: modules/bank/model/bank.class.php [hunk 2]
```diff
@@ -366,178 +374,99 @@ private function bankvotingperiod($Period) {
...
-                $success = true;
-            }
-        }
-        if($InvoiceID && !$success) {
-            if(isset($this->closeablevoucheraccountline['InvoiceID'][$InvoiceID]) && round($this->closeablevoucheraccountline['InvoiceID'][$InvoiceID], 2) == 0) {
-                $success = true;
-            }
+        if(isset($this->closeablevoucheraccountline[$key]) && round($this->closeablevoucheraccountline[$key], 2) == 0) {
+            return true;
         }
```
Comment (mladjo2505) [inline modules/bank/model/bank.class.php:384]: add else return false

File: modules/bank/model/bank.class.php [hunk 3]
```diff
@@ -366,178 +374,99 @@ private function bankvotingperiod($Period) {
...
-        }
-        if($InvoiceID && !$success) {
-            if(isset($this->closeableaccounttilleggline['InvoiceID'][$InvoiceID]) && round($this->closeableaccounttilleggline['InvoiceID'][$InvoiceID], 2) == 0) {
-                $success = true;
-            }
+    private function is_closeable_accounttillegg($AccountPlanID, $KID, $InvoiceID, $JournalID){
+        $key = "B" . $JournalID . "-Fakturanr" . $InvoiceID . "-KID" . $KID;
+        if(isset($this->closeableaccounttilleggline[$key]) && round($this->closeableaccounttilleggline[$key], 2) == 0) {
+            return true;
         }
```
Comment (mladjo2505) [inline modules/bank/model/bank.class.php:412]: add else return false

File: modules/bank/model/bank.class.php [hunk 4]
```diff
@@ -366,178 +374,99 @@ private function bankvotingperiod($Period) {
...
-        }
-        if($InvoiceID && !$success) {
-            if(isset($this->closeablevouchertilbakeline['InvoiceID'][$InvoiceID]) && round($this->closeablevouchertilbakeline['InvoiceID'][$InvoiceID], 2) == 0) {
-                $success = true;
-            }
+    private function is_closeable_vouchertilbake($AccountPlanID, $KID, $InvoiceID, $JournalID){
+        $key = "B" . $JournalID . "-Fakturanr" . $InvoiceID . "-KID" . $KID;
+        if(isset($this->closeablevouchertilbakeline[$key]) && round($this->closeablevouchertilbakeline[$key], 2) == 0) {
+            return true;
         }
```
Comment (mladjo2505) [inline modules/bank/model/bank.class.php:438]: add else return false

**Comment by mladjo2505** [comment]
First round review: Mladen Drobnjaković (@mladjo2505)
Final review with: @stoivo

File: modules/bank/model/bank.class.php [hunk 5]
```diff
@@ -366,178 +374,105 @@ private function bankvotingperiod($Period) {
...
-        if($InvoiceID && !$success) {
-            if(isset($this->closeableaccounttilleggline['InvoiceID'][$InvoiceID]) && round($this->closeableaccounttilleggline['InvoiceID'][$InvoiceID], 2) == 0) {
-                $success = true;
-            }
+    private function is_closeable_accounttillegg($AccountPlanID, $KID, $InvoiceID, $JournalID){
+        $key = "B" . $JournalID . "-Fakturanr" . $InvoiceID . "-KID" . $KID;
+        if(isset($this->closeableaccounttilleggline[$key]) && round($this->closeableaccounttilleggline[$key], 2) == 0) {
+            return true;
+        } else {
+          return false;
```
Comment (agvald) [inline modules/bank/model/bank.class.php]: indent correctly

File: modules/bank/model/bank.class.php [hunk 6]
```diff
@@ -366,178 +374,105 @@ private function bankvotingperiod($Period) {
...
-            }
-        }
-        if($InvoiceID && !$success) {
-            if(isset($this->closeablevoucheraccountline['InvoiceID'][$InvoiceID]) && round($this->closeablevoucheraccountline['InvoiceID'][$InvoiceID], 2) == 0) {
-                $success = true;
-            }
+        if(isset($this->closeablevoucheraccountline[$key]) && round($this->closeablevoucheraccountline[$key], 2) == 0) {
+            return true;
+        } else {
+          return false;
```
Comment (agvald) [inline modules/bank/model/bank.class.php]: indent correctly

File: modules/bank/model/bank.class.php [hunk 7]
```diff
@@ -366,178 +374,105 @@ private function bankvotingperiod($Period) {
...
-        if($InvoiceID && !$success) {
-            if(isset($this->closeablevouchertilbakeline['InvoiceID'][$InvoiceID]) && round($this->closeablevouchertilbakeline['InvoiceID'][$InvoiceID], 2) == 0) {
-                $success = true;
-            }
+    private function is_closeable_vouchertilbake($AccountPlanID, $KID, $InvoiceID, $JournalID){
+        $key = "B" . $JournalID . "-Fakturanr" . $InvoiceID . "-KID" . $KID;
+        if(isset($this->closeablevouchertilbakeline[$key]) && round($this->closeablevouchertilbakeline[$key], 2) == 0) {
+            return true;
+        } else {
+          return false;
```
Comment (agvald) [inline modules/bank/model/bank.class.php]: indent correctly

**Pull request merged**